# Challenge 1

A and B are events of a certain probability space $(\omega, \sigma, P)$ such that $P(A) = 0.3$, $P(B) = 0.6$ and $P(A \cap B) = 0.1$.

Which are false?
* $P(A \cup B) = 0.6$
* $P(A \cap B^{C}) = 0.2$
* $P(A \cap (B \cup B^{C})) = 0.4$
* $P(A^{C} \cap B^{C}) = 0.3$
* $P((A \cap B)^{C}) = 0.9$

* $P(A \cup B) = 0.6$ is FALSE

* $P(A \cup B) = P(A) + P(B) - P(A \cap B) = 0.3 + 0.6 - 0.1 = 0.8$

_________________

* $P(A \cap B^{C}) = 0.2$ is TRUE

* $P(A \cap B^{C}) = 0.3 - 0.1 = 0.2$

________________

* $P(A \cap (B \cup B^{C})) = 0.4$ is FALSE

* $P(A \cap (B \cup B^{C})) = 0.3$

__________

* $P(A^{C} \cap B^{C}) = 0.3$ is FALSE

* $P(A^{C} \cap B^{C}) = 1 - P(A \cup B) = 1 - 0.8 = 0.2$

_________

* $P((A \cap B)^{C}) = 0.9$ is TRUE

* $P((A \cap B)^{C}) = 1 - P(A \cap B) = 1 - 0.1 = 0.9$

# Challenge 2
We have an urn with 10 white balls, 12 red balls and 8 black balls. Calculate the probability of:
* Taking a white ball.
* Taking a white ball after taking a black ball.
* Taking a red ball after taking a black and a red ball.
* Taking a red ball after taking a black and a red ball with reposition (that means after taking a ball I put it back to the urn).


In [18]:
#Taking a white ball.
White_balls=10
Red_balls=12
Black_balls=8
Total = White_balls + Red_balls + Black_balls

White_balls/Total

0.3333333333333333

In [5]:
#Taking a white ball after taking a black ball.
(10/Total) * 8/(Total-1)

0.09195402298850575

In [16]:
#Taking a red ball after taking a black and a red ball.
(Red_balls/Total) * (Black_balls/(Total-1)) * (Red_balls-1)/(Total-2)

0.04334975369458128

In [17]:
#Taking a red ball after taking a black and a red ball with reposition (that means after taking a ball I put it back to the urn).
(Red_balls/Total) * (Black_balls/(Total)) * (Red_balls-1)/(Total)

0.03911111111111111

# Challenge 3

You are planning a picnic today, but the morning is cloudy. So you look for some data:
* 50% of all rainy days start off cloudy!
* But cloudy mornings are common (about 40% of days start cloudy)
* And this is usually a dry month (only 3 of 30 days (10%) tend to be rainy)

What is the chance of rain during the day?



P (Rain) = 0.1

P (Cloudy) = 0.4

P (Rain | Cloudy) = (P (Rain) * P(Cloudy | Rain)) / P (Cloudy)

In [21]:
( 0.1 * 0.5 ) / 0.4

0.125

# Challenge 4

By conducting a telephone survey, we asked 1000 people if they thought it is necessary to have more street lighting at night.

We have met 480 men, 324 of which answered yes and 156 no, as well as 520 women, 351 of which answered yes and 169 no. We wonder if men and women have a different opinion or if gender is irrelevant to the question.

(To see more clearly what they told us, it is best to place the data in a table)


Consider the following events:
- Wants more light (answered yes)
- A man answered


We ask ourselves if these events are independent, that is, if the fact of wanting more light depends on whether one is male or female. Are these events independent or not?

In [22]:
import pandas as pd
import numpy as np

In [23]:
telephone_survey = pd.DataFrame(np.array([[324, 156, 480],[351, 169, 520],[675, 325, 1000]]), columns = ['Yes','No','Total'], index = ['Men','Women','Total'])
telephone_survey

Yes   No  Total
Men    324  156    480
Women  351  169    520
Total  675  325   1000

In [25]:
telephone_survey_prob = telephone_survey/1000
telephone_survey_prob

Yes     No  Total
Men    0.324  0.156   0.48
Women  0.351  0.169   0.52
Total  0.675  0.325   1.00

P (Yes more light) = 0.675

P (Men|Yes More light) = P (Yes More light n Men) / P (Men)

In [27]:
0.324/0.48

0.675

P (Women|Yes More light) = P (Yes More light n Women) / P (Women)

In [29]:
0.351/0.52

0.6749999999999999

**As distribution is the same, events are independent**